server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1460,8 +1460,12 @@ int insertPendingSegmentsIntoMetastore(
         ));
 
     final String now = DateTimes.nowUtc().toString();
+    final Set<String> processedSegmentIds = new HashSet<>();
```

Comment:
Use set of `SegmentIdWithShardSpec` instead.

Reply:
Could you please share why? The string is unique, and also acts as the primary key for the table

Reply:
It makes the code easier to understand. Having a set of `SegmentIdWithShardSpec` would have the exact same effect since the equals and hashCode of `SegmentIdWithShardSpec` uses the same id which is returned in `toString()`.

IIRC, the original code (i.e. pre-`PendingSegmentRecord`) was also maintaining a set of `SegmentIdWithShardSpec`.

Reply:
Done. Thanks for the explanation. I've added a test as well.